File: Data Structures & Algorithms/jump-game-ii/submission-1.java
class Solution {
    public int jump(int[] nums) {
        int l=0,r = 0;
        int res = 0;
        while(r<nums.length-1){
            int farthest = 0;
            for(int i  = l; i<r+1; i++){
                farthest = Math.max(farthest, nums[i] + i);
            }
            l = r+1;
            res++;
            r  = farthest;
        }
        return res;
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-1.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l,r =1,max(piles)
        res = 0

        
        while l<=r:
            hours=0
            m = (l+r)//2
            for n in piles:
                hours += (n + m - 1) // m
                
            if hours>h:
                
                l=m+1
            else:
                res= m
                r=m-1
        return res
            
                 